CMakeLists.txt

```diff
@@ -28,11 +28,11 @@ endif()
 if(__INTEGRATE_OPENCL)
   cmake_minimum_required(VERSION 3.11)
 elseif(USE_GPU OR APPLE)
-  cmake_minimum_required(VERSION 3.2)
+  cmake_minimum_required(VERSION 3.8)
 elseif(USE_CUDA OR USE_CUDA_EXP)
   cmake_minimum_required(VERSION 3.16)
 else()
-  cmake_minimum_required(VERSION 3.0)
+  cmake_minimum_required(VERSION 3.8)
 endif()
```

Comment:
Please change this so that this PR only affects the minimum CMake version required when building the SWIG jar. This change for the benefit of SWIG shouldn't affect the many non-SWIG users of LightGBM.

Something like this:

```cmake
if(__INTEGRATE_OPENCL)
  cmake_minimum_required(VERSION 3.11)
elseif(USE_SWIG)
  cmake_minimum_required(VERSION 3.8)
elseif(USE_GPU OR APPLE)
  cmake_minimum_required(VERSION 3.2)
elseif(USE_CUDA OR USE_CUDA_EXP)
  cmake_minimum_required(VERSION 3.16)
else()
  cmake_minimum_required(VERSION 3.0)
endif()
```